Add docs for unsupported operations and replication (#20896)
supabase/supabase: apps/docs/pages/guides/database/postgres/setup-replication-external.mdx

import { createDecipheriv } from 'crypto'
import Layout from '~/layouts/DefaultGuideLayout'

export const meta = {
  title: 'Replicate to another Postgres database using Logical Replication',
  description:
    'Example to setup logical replication using publish-subscribe to a Postgres database outside of Supabase',
  footerHelpType: 'postgres',
}

For this example, you will need:

- A Supabase project
- A PostgreSQL database (running v10 or newer)

> We will be running commands on both of these databases to publish changes from our Supabase database to our external database.

Reviewer: ```suggestion
You will be running commands on both of these databases to publish changes from the Supabase database to the external database.
```

> 1. We need to create our `publication` so let's run this on our **Supabase database**:

Reviewer: ```suggestion
1. Create a `publication` on the **Supabase database**:
```

```sql
CREATE PUBLICATION example_pub;
```

> 2. Also on our **Supabase database**, let's create a `replication slot`:

Reviewer: ```suggestion
2. Also on the **Supabase database**, create a `replication slot`:
```

```sql
select pg_create_logical_replication_slot('example_slot', 'pgoutput');
```

> 3. Now we will connect to our **external database** and subscribe to our `publication` Note: This will need a direct connection to your database and you can find the connection info in the [Dashboard](https://supabase.com/dashboard/project/_/settings/database)):

Reviewer: ```suggestion
3. Connect to the **external database** and subscribe to the `publication` Note: This will need a direct connection to your database. You can find the connection info in the [Dashboard](https://supabase.com/dashboard/project/_/settings/database):
```

```sql
CREATE SUBSCRIPTION example_sub

> CONNECTION 'host=db.oaguxblfdassqxvvwtfe.supabase.co user=postgres password=YOUR_PASS dbname=postgres'

Reviewer: Needing a direct connection means it can't be done with the pooler even in session mode? We may need an admonition about IPv6 then

Author: Good point, will add. Yep, replication must be direct (although pg16 will allow it from replicas)

PUBLICATION example_pub
WITH (copy_data = true, create_slot=false, slot_name=example_slot);
```

<Admonition type="info">

> We have set `create_slot` to `false` because we have already made a `replication slot` on our **Supabase database** so we provide the name in `slot_name`.

Reviewer: ```suggestion
`create_slot` is set to `false` because `slot_name` is provided and the slot was already created in Step 2.
```

> If we want to copy all the data that existed before the slot was created, then we set `copy_data` to `true`

Reviewer: ```suggestion
To copy data from before the slot was created, set `copy_data` to `true`.
```

</Admonition>

> 4. Now we have a `publication` and a `subscription`, we can add all of the tables to the publication and we are done.

Reviewer: ```suggestion
4. Add all the tables that you want replicated to the publication.
```

```sql
ALTER PUBLICATION example_pub ADD TABLE example_table;
```

> 5. We can check the replication status using `pg_stat_replication`

Reviewer: ```suggestion
5. Check the replication status using `pg_stat_replication`
```
